CorsixTH/Lua/entities/patient.lua

```diff
@@ -245,16 +249,17 @@ function Patient:treated() -- If a drug was used we also need to pay for this
       if self.is_emergency then
         self.hospital.emergency.cured_emergency_patients = hospital.emergency.cured_emergency_patients + 1
       end
+
       self.cured = true
       self.infected = false
-      self:setMood("cured", "activate")
-      self.world.ui:playSound "cheer.wav" -- This sound is always heard
       self.attributes["health"] = 1
-      self:changeAttribute("happiness", 0.8)
-      hospital:changeReputation("cured", self.disease)
-      self.treatment_history[#self.treatment_history + 1] = _S.dynamic_info.patient.actions.cured
-      self:goHome(self.cured)
       self:updateDynamicInfo(_S.dynamic_info.patient.actions.cured)
+      self.hospital:msgCured()
+
+      -- If the patient refused to pay they are on the way home already
+      if not self.going_home then
```

Comment:
It's a good point. That would be more consistent.
Maybe `Hospital:receiveMoneyForTreatment(patient)` should return a boolean, so we know whether the patient paid or not and hence for which reason he's sent home.
Also, I'm wondering if statistic handling (`hospital.num_cured + 1`) shouldn't be managed in one place only (in `goHome()` only and not in `treated()` anymore).